src/logic/groups.js

```diff
@@ -710,14 +710,28 @@ const availabilityRate = (wcif, activity, competitor) => {
   return -(timeWhenBusy / activityDuration(activity));
 };
 
-const overlapsEveryoneWithSameRole = (wcif, groups, activity, competitor) =>
-  intersection(
+const overlapsEveryoneWithSameRole = (wcif, groups, activity, competitor) => {
+  const roleMapping = {
+    'trainee-delegate': 'delegate',
+  };
+
+  const normalizedRoles = competitor.roles.map(
+    role => roleMapping[role] || role
+  );
+
+  return intersection(
     ['staff-dataentry', 'delegate', 'trainee-delegate', 'organizer'],
-    competitor.roles
```

Comment:
What I would do is move it to a function:

```js
function normalizeRoles(roles) {
  // Treat trainee delegates the same as delegates.
  const mapping = { 'trainee-delegate': 'delegate' }
  return roles.map(role => mapping[role] || role);
}
```

And then in places that we use `competitor.roles` (or `person.roles`) we just wrap it as `normalizeRoles(competitor.roles)`.

We should do that wherever `.roles` is used in this file, except places that do explicit `.roles.includes` check.